include/seqan3/alignment/multiple/detail/align_multiple_seqan2_adaptation.hpp

```diff
@@ -0,0 +1,285 @@
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides some detail functionality to the algorithm seqan3::align_multiple.
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ * \author Joerg Winkler <j.winkler AT fu-berlin.de>
+ * \author Simon Sasse <simon.sasse AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/std/algorithm>
+#include <cstdlib>
+#include <string>
+#include <vector>
+
+#include <seqan3/core/platform.hpp>
+
+#if SEQAN3_HAS_SEQAN2
+#include <seqan/basic.h>
+#include <seqan/graph_msa.h>
+#include <seqan/modifier.h>
+#include <seqan/reduced_aminoacid.h>
+#include <seqan/seq_io.h>
+#include <seqan/stream.h>
+#endif
+
+#include <seqan3/alignment/configuration/align_config_band.hpp>
+#include <seqan3/alignment/configuration/align_config_gap.hpp>
+#include <seqan3/alignment/configuration/align_config_scoring_scheme.hpp>
+#include <seqan3/alignment/scoring/nucleotide_scoring_scheme.hpp>
+#include <seqan3/alphabet/aminoacid/all.hpp>
+#include <seqan3/alphabet/gap/gapped.hpp>
+#include <seqan3/alphabet/nucleotide/all.hpp>
+#include <seqan3/core/algorithm/configuration.hpp>
+#include <seqan3/core/detail/test_accessor.hpp>
+#include <seqan3/core/type_list/traits.hpp>
+#include <seqan3/core/type_traits/template_inspection.hpp>
+#include <seqan3/range/views/char_to.hpp>
+#include <seqan3/range/views/chunk.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief Holds all functionality to make the seqan3::align_multiple interface compatible with the SeqAn2 msa algorithm.
+ * \ingroup multiple_alignment
+ * \sa seqan3::align_multiple
+ */
+template <typename seqan3_alphabet_type>
+class align_multiple_seqan2_adaptation
+{
+private:
+    //!\brief A list of SeqAn3 types that are currently supported for seqan3::align_multiple.
+    using seqan3_types = type_list<dna4, dna5, dna15, rna4, rna5, aa27, aa10murphy, aa10li>;
+    //!\brief The corresponding list of SeqAn2 types (the order must be the same!).
+    using seqan2_types = type_list<seqan::Dna,
+                                   seqan::Dna5,
+                                   seqan::Iupac,
+                                   seqan::Rna,
+                                   seqan::Rna5,
+                                   seqan::AminoAcid,
+                                   seqan::ReducedAminoAcid<seqan::Murphy10>,
+                                   seqan::ReducedAminoAcid<seqan::Li10>>;
+    //!\brief The index of the seqan3_alphabet_type in the list of seqan3_types.
+    static constexpr auto index = list_traits::find<seqan3_alphabet_type, seqan3_types>;
```

Comment:
Alright, sounds like a good plan. But it also sounds like this might take some time because we can't estimate how easy it will be to adapt the interfaces. I would take this on as a follow up that we can discuss and schedule afterwards if that is fine?

Reply:
I made a PR against your repo branch a while ago. There it is already removed and done via `char` only. Have you seen it?